itests/mir_test.go

```diff
@@ -74,26 +59,88 @@ func runMirManglingTests(t *testing.T, opts ...interface{}) {
 	t.Run("testMirMiningWithMessaging", ts.testMirAllNodesMiningWithMessaging)
 }
 
-func runMirConsensusTests(t *testing.T, opts ...interface{}) {
-	ts := itestsConsensusSuite{opts: opts}
-
+func TestMirOneNodeMining(t *testing.T) {
+	ts := itestsConsensusSuite{opts: []interface{}{kit.ThroughRPC(), kit.MirConsensus()}}
```

Comment:
It looks like this itestsConsensusSuite is (1) immutable and (2) the same for all tests. I think we can easily avoid all the boilerplate in these tests. First, define `ts` as a package-level variable (alternatively, you can use `TestMain` or `init()` to initialize `ts` but it seems unnecessary and will require you to move this file to its own package). Second, get rid of the `t.Run()` calls and just inline the test logic in the `TestMir*` methods directly.

Reply:
In fact, I don't think you need the `iTestsConsensusSuite` struct at all. Just define at package-level:

`var mirTestOpts = []interface{}{kit.ThroughRPC(), kit.MirConsensus()}`

and pass it to the ensemble calls.

Reply:
testMir* methods are reused in tests with mangler.

Reply:
My suggestion there is to define `TestMir*WithMangler` test methods that wrap the `TestMir*` test functions with setting/unsetting that environment variable. Alternatively, the test functions that we want to run with the mangler environment variable, can be rewritten as a form of table-driven tests.

Reply:
Ok. I will do that in another PR. Thank you @hmoniz.